static/app/components/u2f/u2finterface.tsx

```diff
@@ -169,9 +193,19 @@ class U2fInterface extends React.Component<Props, State> {
         this.submitU2fResponse(promise);
       }
     } else if (this.props.flowMode === 'enroll') {
-      const {registerRequests, registeredKeys} = this.props.challengeData;
-      promise = u2f.register(registerRequests as any, registeredKeys as any);
-      this.submitU2fResponse(promise);
+      const {organization} = this.props;
+      if (organization.features.includes('webauthn-register')) {
+        const challengeArray = base64urlToBuffer(
+          this.props.challengeData.webAuthnRegisterData
+        );
+        const challenge = cbor.decodeAllSync(challengeArray);
```

Comment:
was there a lost comment here about making this async?

Reply:
yes, will make this async

Reply:
i tried making it async, but it seems the promise is still pending at the next line and the challenge is needed so it errors out. How should i go about this? and is it bad to leave this as a sync fn?

Reply:
@maxiuyuan did you await on it, or put the rest of the code into a `.then`? (probably good to read the docs on promises again https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

I'm not sure that it matters since from what I understand it will only be non-blocking if the work is put in a worker, which I don't think the library does (but worth trying anyway)